python/invoke_release/tasks.py

```diff
@@ -398,9 +398,20 @@ def _prompt_for_changelog(verbose):
             _verbose_output(verbose, 'User has closed editor')
 
             with codecs.open(tf.name, 'rb', encoding='utf8') as read:
+                first_line = True
+                last_line_blank = False
                 for line in read:
-                    if line and line.strip() and not line.startswith(CHANGELOG_COMMENT_FIRST_CHAR):
+                    line_blank = not line.strip()
+                    if (first_line or last_line_blank) and line_blank:
+                        # Suppress leading blank lines and compress multiple blank lines into one
+                        continue
+                    if not line.startswith(CHANGELOG_COMMENT_FIRST_CHAR):
                         changelog_message.append(line)
+                        last_line_blank = line_blank
+                        first_line = False
```

Comment:
should these two lines be inside the _if_?

What would be expected as an output for:
```
# foo

# bar

# baz

msg
```

Reply:
How about skipping comments as the very first thing?
```
for line in read:                  
    if line.startswith(CHANGELOG_COMMENT_FIRST_CHAR):
        continue
    line_blank = not line.strip()
    if (first_line or last_line_blank) and line_blank:
        # Suppress leading blank lines and compress multiple blank lines into one
        continue

    changelog_message.append(line)
    last_line_blank = line_blank
    first_line = False
```

Reply:
So, we don't want to skip comments as the very first thing, but we can test the inverse and continue on comment to de-dent those last three lines of the loop. Basically, given an input like this:

```

This is a note or something


- Hello
- Goodbye

# This is a comment

- One more

# These are the instruction comments at the end of the VIM window
```

We want the output to be:

```
This is a note or something

- Hello
- Goodbye

- One more
```

Notice that there are no repeat blank lines, but otherwise blank lines are preserved. If we moved the comment skip to the top of the loop, we'd still have two blank lines where the comment was removed.